services/weatherService.js

```diff
@@ -0,0 +1,67 @@
+fluid.registerNamespace("myL3");
+
+fluid.defaults("myL3.weatherService", {
+    gradeNames: "fluid.component",
+    invokers: {
+        addDataIntoTable: {
+            funcName: "myL3.weatherService.addDataIntoTable",
+            args: ["{arguments}.0", "{arguments}.1"]
+        }
+    }
+});
+
+myL3.weatherService.addDataIntoTable = function (weatherData, fields) {
```

Comment:
Please add documentation on the format of the function parameters.

And please also change the structure of the `fields` parameter. From looking at the code, I believe that it is an array of integers, where each integer is the id of a selected data field from the selector dialog box. This is very opaque and brittle.

The purpose of this function is to populate an HTML table, given weather data and a selected set of fields. A better model for the fields would be a set of boolean flags, such as:

```
{
    tempC: true,
    humidity: true
}
```

This is now no longer tied to the UI of the dialog box and is understandable when inspecting (in the current implementation, the equivalent would be the much more opaque: `[2, 4]`).